backend/src/case_file/case_file.service.ts

```diff
@@ -906,8 +1027,435 @@ export class CaseFileService {
+        action_code: true,
+        action_type_action_xref_guid: true,
+        action_code_action_type_action_xref_action_codeToaction_code: {
+          select: {
+            short_description: true,
+            long_description: true,
+            active_ind: true,
+          },
+        },
+      },
+    });
+
+    // construct the equipmentDetails list
+    for (const action of actions) {
+      const equipment = action.equipment;
+
+      // get the action xref for the action
+      let actionData = actionCodes.find(
+        (element) =>
+          element.action_type_action_xref_guid ===
+          action.action_type_action_xref_guid
+      );
+
+      if (equipment && equipment.active_ind) {
+        // prisma doesn't support the geometry type, for now, it's just treated as a string
+        // Parse the geometry string into a GeoJSON object
+
+        // Correctly setting the search path using Prisma
+        await this.prisma
+          .$executeRaw`SET search_path TO public, case_management`;
+
+        // get the latitude and longitude using a raw query
+        const result = await this.prisma.$queryRaw<
+          { longitude: number; latitude: number }[]
+        >`
+            SELECT 
+              public.st_x(equipment_geometry_point::geometry) AS longitude, 
+              public.st_y(equipment_geometry_point::geometry) AS latitude
+            FROM 
+              ${Prisma.raw("case_management.equipment")}
+            WHERE 
+              equipment_guid = ${Prisma.raw(
+                `'${equipment.equipment_guid}'::uuid`
+              )}
+          `;
+
+          const { longitude, latitude } = result[0];
+
+          const longitudeString = longitude ? `${longitude}` : null;
+          const latitudeString = longitude ? `${latitude}` : null;
+
+          let equipmentDetail =
+            equipmentDetailsMap.get(equipment.equipment_guid) ||
+            ({
+              equipmentGuid: equipment.equipment_guid,
+              equipmentTypeCode: equipment.equipment_code,
+              equipmentTypeActiveIndicator: equipment.active_ind,
+              address: equipment.equipment_location_desc,
+              xCoordinate: longitudeString,
+              yCoordinate: latitudeString,
+              actions: [],
+            } as Equipment);
+
+          // Append the action to this equipment's list of actions
+          equipmentDetail.actions.push({
+            actionGuid: action.action_guid,
+            actor: action.actor_guid,
+            date: action.action_date,
+            activeIndicator: action.active_ind,
+            actionCode: actionData.action_code,
+          });
+
+          equipmentDetailsMap.set(equipment.equipment_guid, equipmentDetail);
+      }
+    }
+    const equipmentDetails = Array.from(
+      equipmentDetailsMap.values()
+    ) as Equipment[];
+
+    return equipmentDetails;
+  }
+
+  async deleteEquipment(
+    deleteEquipmentInput: DeleteEquipmentInput
+  ): Promise<boolean> {
```

Comment:
I'll leave this one up to you, but I think it would be best to return the caseFile rather than a boolean. It would also be consistent as well